public/static/app/common/ApiUtil.js

```diff
@@ -1,19 +1,41 @@
-mciModule.factory('ApiUtil', function($http) {
+mciModule.factory("ApiUtil", function ($http) {
   // TODO Pass CSRF Token (currently, all queries use GET method)
   // TODO Use $httpProvider (tech-debt ticket required)
   return {
-    httpGetter: function(base) {
-      let realBase = base || '';
-      realBase = realBase.endsWith('/') ? realBase.substr(0,realBase.length - 1): realBase;
+    httpGetter: function (base) {
+      let realBase = base || "";
+      realBase = realBase.endsWith("/")
+        ? realBase.substr(0, realBase.length - 1)
+        : realBase;
 
-      return function(apiEndpoint, endpointTplParams, httpParams) {
-        if (apiEndpoint[0] !== '/') apiEndpoint = '/' + apiEndpoint;
+      return function (apiEndpoint, endpointTplParams, httpParams) {
+        if (apiEndpoint[0] !== "/") apiEndpoint = "/" + apiEndpoint;
         return $http.get(
           // Interpolate endpoint template with params
           realBase + _.template(apiEndpoint)(endpointTplParams),
-          {params: httpParams, withCredentials: true}
-        )
-      }
-    }
-  }
+          { params: httpParams, withCredentials: true }
+        );
+      };
+    },
+    httpPoster: function (base) {
+      let realBase = base || "";
+      realBase = realBase.endsWith("/")
+        ? realBase.substr(0, realBase.length - 1)
+        : realBase;
+
+      return function (apiEndpoint, endpointTplParams, body) {
+        if (apiEndpoint[0] !== "/") apiEndpoint = "/" + apiEndpoint;
+        data = JSON.stringify(body);
```

Comment:
I believe $http accepts an object in the data param so you don't need to do this

Reply:
Because evergreen is hosted outside of the ELB for Kanopy, every request we send has to have our corp secure cookie on it. As well, these requests have to deal with CORS because it's a different origin. When you do anything but a simple request (gets, posts with a content-type of text/plain), a preflight OPTIONS request is sent. `$http` doesn't send the corp secure cookie on these preflight requests so the ELB redirects them to login and the overall request fails.

By setting the content-type to text/plain explicitly and stringifying the body, we avoid these preflight requests and the cookie gets set correctly.

The other option here is to work with the Kanopy team and see about not requiring cookies on OPTION requests as technically they're not required. That's being tracked [here](https://jira.mongodb.org/browse/DEVOPS-1812). This lets us get around the issue in the meantime.